File: qc-application/src/main/java/com/eager/questioncloud/application/api/feed/subscribe/service/FeedSubscribeService.java
package com.eager.questioncloud.application.api.feed.subscribe.service;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.subscribe.dto.SubscribeDetail;
import com.eager.questioncloud.core.domain.subscribe.infrastructure.repository.SubscribeRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FeedSubscribeService {
    private final SubscribeRepository subscribeRepository;

    public List<SubscribeDetail> getMySubscribes(Long userId, PagingInformation pagingInformation) {
        return subscribeRepository.getMySubscribes(userId, pagingInformation);
    }

    public int countMySubscribe(Long userId) {
        return subscribeRepository.countMySubscribe(userId);
    }
}

File: qc-application/src/main/kotlin/com/eager/questioncloud/application/api/feed/subscribe/service/FeedSubscribeService.kt
package com.eager.questioncloud.application.api.feed.subscribe.service

import com.eager.questioncloud.application.api.creator.dto.CreatorInformation
import com.eager.questioncloud.application.api.creator.implement.CreatorInformationReader
import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.subscribe.infrastructure.repository.SubscribeRepository
import org.springframework.stereotype.Service

@Service
class FeedSubscribeService(
    private val subscribeRepository: SubscribeRepository,
    private val creatorInformationReader: CreatorInformationReader,
) {
    fun getMySubscribes(userId: Long, pagingInformation: PagingInformation): List<CreatorInformation> {
        val subscribedCreatorIds = subscribeRepository.getMySubscribedCreators(userId, pagingInformation)
        return creatorInformationReader.getCreatorInformation(subscribedCreatorIds)
    }

    fun countMySubscribe(userId: Long): Int {
        return subscribeRepository.countMySubscribe(userId)
    }
}
